### ISSUE TITLE
Handle errors for submitToBasket and prevent a form-action when subscribing to newsletters

### ISSUE THREAD

**Issue by meandavejustice** [mozilla/testpilot#3694]
```
Content Security Policy: The page’s settings blocked the loading of a resource at https://example.com:8000/ (“form-action”).
```

* mobileDialog
* newsletterForm/newsletterFooter

**Comment by meandavejustice** [comment]
There's been a lot of time spent on this so far with no update to the issue, so here's some info.

Based on what we know about how `event.preventDefault` works with forms, the current code _should_ be working correctly to block any sort of submission. It clearly is not because we are getting the above error.

The issue seems to be related to react's event handling. One source said that React's event handlers are triggered at the root of the react app, so for events like `onsubmit`, the handler is called too late.
In an attempt to address this issue we tried attaching a native event handler to the actual element via a `ref`. This created a puzzling issue where the error was logged before the handler was event called (previously the handler was called, but `preventDefault` was ignored.

Many other tweak were tried to make this work, including `event.nativeEvent.stopImmediatePropagation`. Nothing I do seems to work here.

I'm going to focus on some other issues for a bit, as I feel we've spent too much time on this as is. This issue doesn't affect users as the form submission is still handled correctly. The issue is only about the CSP error log.

It may make sense to move this into the stretch milestone.

**Comment by fzzzy** [comment]
I agree we should move this to stretch or just close it. We're trying to prevent the form from posting and are unable to do so because of the way react works, and none of the workarounds we tried helped. CSP is blocking the form post instead, which is good; so the only bad thing is that CSP prints an ugly error in the console when it blocks it. This isn't important enough to spend as much time as we have spent on it.
